File: spec/factories/game.rb
FactoryBot.define do
  factory :game do
    association :league
    association :split

    date do 100.years.ago end
    winner_id do 1 end
    loser_id do 1 end
    blue_side_team_id do 1 end
    red_side_team_id do 2 end
    game_number do 1 end
  end

  trait :with_teams do
    after :create do |game|
      blue_team = create(:blue_side_team)
      red_team = create(:red_side_team)

      game.blue_side_team_id = blue_team.id
      game.red_side_team_id = red_team.id
    end
  end

  trait :unplayed do
    after :create do |game|
      game.update_attributes!(winner_id: nil, loser_id: nil)
    end
  end

  trait :with_stats do
    after :create do |game|
      game.blue_side_team_fb_odds = 1.5
      game.blue_side_team_fd_odds = 1.5
      game.blue_side_team_ft_odds = 1.5
      game.blue_side_team_fbaron_odds = 1.5
      game.blue_side_team_win_odds = 1.5

      game.red_side_team_fb_odds = 2.5
      game.red_side_team_fd_odds = 2.5
      game.red_side_team_ft_odds = 2.5
      game.red_side_team_fbaron_odds = 2.5
      game.red_side_team_win_odds = 2.5

      game.first_blood_team_id = game.blue_side_team.id
      game.first_turret_team_id = game.blue_side_team.id
      game.first_dragon_team_id = game.blue_side_team.id
      game.first_baron_team_id = game.blue_side_team.id

      game.winner_id = game.red_side_team.id
      game.loser_id = game.blue_side_team.id

      game.save!
    end
  end
end

File: analysis/schema/game.py
from datetime import datetime
from peewee import *

from schema.base import BaseModel

class Game(BaseModel):
    from schema.teams import Team
    from schema.split import Split
    from schema.league import League

    id = PrimaryKeyField()
    red_side_team = ForeignKeyField(Team, column_name='red_side_team_id')
    blue_side_team = ForeignKeyField(Team, column_name='blue_side_team_id')

    red_side_team_id = IntegerField(Team)
    blue_side_team_id = IntegerField(Team)
    first_blood_team_id = IntegerField(Team)
    first_turret_team_id = IntegerField(Team)
    first_dragon_team_id = IntegerField(Team)
    first_baron_team_id = IntegerField(Team)

    blue_side_team_fb_odds = DoubleField()
    blue_side_team_fd_odds = DoubleField()
    blue_side_team_ft_odds = DoubleField()
    blue_side_team_fbaron_odds = DoubleField()

    red_side_team_fb_odds = DoubleField()
    red_side_team_fd_odds = DoubleField()
    red_side_team_ft_odds = DoubleField()
    red_side_team_fbaron_odds = DoubleField()

    date = DateTimeField()
    split = ForeignKeyField(Split)
    league = ForeignKeyField(League)

    class Meta:
        table_name = 'games'

    @classmethod
    def previous_n_regular_season_games_for_team(cls, team, n, game_date):
        games =  [
                g for g in (cls
                    .select()
                    .where(
                        (
                            (cls.blue_side_team_id == team.id) |
                            (cls.red_side_team_id == team.id)
                        ) &
                        (
                            Game.date < game_date
                        ) &
                        (
                            (
                                (Game.date > datetime(2019, 1, 1))
                                # remove to exclude playoff games from prev. split
                                & (Game.date < datetime(2019, 3, 30))
                            ) |
                            (
                                (Game.date > datetime(2019, 6, 1)) &
                                (Game.date < datetime(2019, 8, 8))
                            )
                        )
                    )
                    .limit(n)
                    .order_by(cls.date.desc())
                    )
                ]

        games.reverse()
        return games
